src/app/containers/visualizer/components/swt-surface-plot/swt-surface-plot.component.ts

```diff
@@ -58,14 +59,15 @@ export class SwtSurfacePlotComponent implements OnChanges, OnInit {
     }
 
     drawAltitudeBox() {
+        // remove any existing altitude boxes
+        this.g2.selectAll('#altitude-box').remove();
         const geoBox: d3.GeoPermissibleObjects = this.geoBoxFromPoint(this.longitude, this.latitude);
         // draw a red box around the altitude profile location
-        this.g2.selectAll('#altitude-box')
-            .data([ geoBox ]) // feature
-            .enter()
```

Comment:
So, I think you want to leave the `.enter().append()` if you can. That is d3 working it's magic under the hood. I think the idea is that it will take care of all the addition/deletion/modification of the DOM features. The biggest benefit of leaving them in I saw before was that it doesn't cause the blinking from delete/remake.

Here is a quick observable I found that describes some of the joins. I saw a really good one before, but couldn't find it again.
https://observablehq.com/@dnarvaez27/understanding-enter-exit-merge-key-function

Reply:
Oh, okay, cool explanation. Let's use them. Except for some reason `.enter()` doesn't work with the altitude box (I'm sure it _could_ work), but this is a case where there is only ever one box at a time, I don't think there is blinking/flashing, and I don't think it matters so much. Let me know if you want me to refactor or if it works for you as is.

Reply:
Yeah, I think this one box is OK the way you have it, and you're only calling it when it needs to be updated not every other redraw call.